lieer/gmailieer.py

```diff
@@ -707,21 +708,38 @@ def send (self, args):
     self.setup (args, args.dry_run, True, True)
     self.remote.get_labels ()
 
-    if args.message == '-':
-      msg = sys.stdin.buffer.read()
-      fn = 'stdin'
-    else:
-      if os.path.isabs(args.message):
-        fn = args.message
-      else:
-        fn = os.path.join(self.cwd, args.message)
-      msg = open(fn, 'rb').read()
+    msg = sys.stdin.buffer.read()
+    fn = 'stdin'
 
     # check if in-reply-to is set and find threadId
     threadId = None
 
     import email
     eml = email.message_from_bytes(msg)
+
+    # If there are recipients passed on the CLI, we need to compare them with
+    # what's in the message headers, as they need to match the message body
+    # (we can't express other recipients via the GMail API)
+
+    cli_recipients = set(args.recipients)
+
+    # construct existing recipient address list from To, Cc, Bcc headers
+    header_recipients = set()
+    for field_name in ("To", "Cc", "Bcc"):
+      field_values = eml.get_all(field_name, [])
+      field_addrs = map(lambda x: email.utils.parseaddr(x)[1], field_values)
+      header_recipients = header_recipients.union(field_addrs)
+
+    if args.read_recipients:
+      if not header_recipients.issuperset(cli_recipients):
+          print ("Recipients passed via CLI, but not part of message headers: {}".format(", ".join(cli_recipients.difference(header_recipients))))
+          exit (2)
+    elif not header_recipients == cli_recipients:
+      print ("Recipients passed via CLI differ from those in message headers:")
+      print (" - {}".format(", ".join(cli_recipients)))
+      print (" - {}".format(", ".join(header_recipients)))
+      exit (2)
```

Comment:
Same here.

Reply:
I now `raise argparse.ArgumentError` with the (condensed) error message.